src/stages/downloader.rs

```diff
@@ -29,44 +36,14 @@ where
     where
         'db: 'tx,
     {
-        let _ = tx;
         let past_progress = input.stage_progress.unwrap_or_default();
 
-        if !input.restarted {
-            info!("Waiting for headers...");
-            let dur = Duration::from_millis(rand::thread_rng().gen_range(3000..6000));
-            sleep(dur).await;
-        }
-
-        info!("Processing headers");
-
-        let target = past_progress + 100;
-
-        let commit_block = rand::random::<bool>()
-            .then(|| past_progress + rand::thread_rng().gen_range(0..*target));
-
-        let mut processed = past_progress;
-        let mut must_commit = false;
-        for block in past_progress..=target {
-            info!(block = block.0, "(mock) Downloading");
-
-            processed.0 += 1;
-
-            if let Some(commit_block) = commit_block {
-                if block == commit_block {
-                    must_commit = true;
-                    break;
-                }
-            }
+        self.downloader.run(None, tx).await?;
 
-            let dur = Duration::from_millis(rand::thread_rng().gen_range(0..500));
-            sleep(dur).await;
-        }
-        info!(highest = target.0, "Processed");
         Ok(ExecOutput::Progress {
-            stage_progress: processed,
-            done: !must_commit,
-            must_commit,
+            stage_progress: past_progress,
```

Comment:
Per stagedsync API, downloader must accept `past_progress` as starting point and must return `stage_progress` where it reached on current stage run.

The way it's done here, the stage will infinitely loop.

Reply:
Yes, I understand that. The feature to continue from the last point will be implemented in a subsequent PR. I've commented out the stage from the akula main binary so that it is not disturbing you for now.